proof-of-us-v1/proof-of-us.pact

```diff
@@ -183,17 +176,22 @@
     (create-token-id { 'uri: uri, 'precision: 0, 'policies: TOKEN-POLICIES } (create-capability-guard (CONNECT event-id uri)))
   )
 
-  (defun create-and-mint-connection-token:string (event-id:string uri:string connection-guards:[guard])
+  (defun create-and-mint-connection-token:string (uri:string connection-guards:[guard])
     (enforce (>= (length connection-guards) 2) "At least 2 connections are required to mint a connection token")
-    (validate-collection event-id (read-msg COLLECTION-ID-MSG-KEY))
-
+    
     (let* (
+      (event-id:string (try "" (read-msg EVENT-ID-MSG-KEY)))
       (creator-guard:guard (create-capability-guard (CONNECT event-id uri)))
       (token-id (create-token-id { 'uri: uri, 'precision: 0, 'policies: TOKEN-POLICIES } creator-guard)))
 
+      (if (= event-id "") true [
+          (validate-event-collection event-id (read-msg COLLECTION-ID-MSG-KEY))
+          (validate-event event-id)
+        ]
+      )
       (with-capability (CONNECT event-id uri)
-        (map (enforce-pou-guard) connection-guards)
-        (with-capability (TOKEN_CREATION event-id)
+        (with-capability (INTERNAL token-id)
```

Comment:
Can we do the connection-guards enforcement before bringing the 'INTERNAL' cap in scope? Might not change much here but it's good practice to have the scope of a cap as small as possible.

Reply:
Done! 😄